### PR TITLE
Dev 2.12

### PR DESCRIPTION
**Pull request by taojing2002**

### CODE REVIEW

File: test/edu/ucsb/nceas/metacat/dataone/D1NodeServiceTest.java [hunk 1]
```diff
@@ -61,11 +65,15 @@
 import org.dataone.service.types.v2.util.ObjectFormatServiceImpl;
 import org.dataone.service.util.Constants;
 import org.dataone.service.util.TypeMarshaller;
+import org.mockito.Mockito;
```
[the remark this answers is not in the document]
Comment (taojing2002) [reply test/edu/ucsb/nceas/metacat/dataone/D1NodeServiceTest.java]: Yeah. I found it is helpful and started to use it.

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterProcessor.java [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+ * For each matching 'filter' element in a document, build a query term from it. These individual
+ * query terms will assembled into a complete query by calling routine.
+ *
+ * Each filter type can have a set of leaf elements that will be used to build the search term
+ * contributed by that filter to the entire query being assembled.
+ * Each filter can have parameters defined that control how the query is built.
+ *
+ * @author slaughter
+ *
+ */
```
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterProcessor.java]: OK, will do.

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/FilterCommonRootSolrField.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+/**
+ * A complex data value mining SolrField. This class returns a value to
+ * an indexing subprocessor from its dependent class. See FilterRootElement for
+ * a typical usage.
+ *
+ * @author slaughter
```
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/FilterCommonRootSolrField.java]: ok

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+/**
+ * Used by FilterCommonRootSolrField.
+ *
+ * Assembled a query string based on a set of filters in a DataONE collection document.
+ *
+ * @author slaughter
+ *
+ * Based on CommonRootSolrField by sroseboo
+ *
+ */
```
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java]: yes

File: src/edu/ucsb/nceas/metacat/dataone/D1NodeService.java [hunk 5]
```diff
@@ -2216,11 +2219,50 @@ public InputStream view(Session session, String format, Identifier id)
...
+   * stylesheet.parameters.1.name=serverName
+   * stylesheet.parameters.1.value=server.name
+   * stylesheet.parameters.2.name=organization
+   * stylesheet.parameters.2.value=ESS-DIVE
+   * The value can be either a name of metacat properties (e.g.server.name) or a real value (e.g. ESS-DIVE).
+   * @return the params with more name/value pairs from the theme properties file.
+ * @throws ServiceException 
+ * @throws PropertyNotFoundException 
+   */
+  static void addParasFromSkinProperties(Hashtable<String, String[]> params, String format) throws ServiceException, PropertyNotFoundException {
```
[the remark this answers is not in the document]
Comment (taojing2002) [reply src/edu/ucsb/nceas/metacat/dataone/D1NodeService.java]: Fixed it. See the commit 99b37f773c1b0ff92cb4f8d552b96c450e14843f

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java]: fixed in commit 706e9aeb23923213545e851383a0c29436166a2f

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterRootElement.java]: fixed in commit d7050743507a73b2660b3a6fd54a1ea3f59b11e1

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/utility/FilterProcessor.java]: fixed in commit d7050743507a73b2660b3a6fd54a1ea3f59b11e1  and d7050743507a73b2660b3a6fd54a1ea3f59b11e1

File: metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/FilterCommonRootSolrField.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gothub) [reply metacat-index/src/main/java/edu/ucsb/nceas/metacat/index/parser/FilterCommonRootSolrField.java]: fixed in commit d7050743507a73b2660b3a6fd54a1ea3f59b11e1

**Pull request merged**